controllers/rule_controller.go

```diff
@@ -46,12 +45,14 @@ type RuleReconciler struct {
 	Log              logr.Logger
 	RuleConfigmap    types.NamespacedName
 	ValidationConfig validation.Config
+	RulesFileName    string
```

Comment:
I wouldn't use "RulesFileName" here. Controller actually doesn't care if this map is mounted anywhere - it's domain is to create a valid ConfigMap. `dataKey` or `mapKey` should be sufficient.